src/async-wrap.h

```diff
@@ -46,10 +47,15 @@ class AsyncWrap : public BaseObject {
 #undef V
   };
 
-  inline AsyncWrap(Environment* env,
-                   v8::Local<v8::Object> object,
-                   ProviderType provider,
-                   AsyncWrap* parent = nullptr);
+  AsyncWrap(Environment* env,
+            v8::Local<v8::Object> object,
+            ProviderType provider,
+            AsyncWrap* parent = nullptr);
+
+  AsyncWrap(v8::Isolate* isolate,
+            v8::Local<v8::Object> object,
+            ProviderType provider,
```

Comment:
Yes. Users don't have access to `Environment`. It's the same reason a chunk of node's Buffer API are in `node_internals.h` that also accept `Environment` instead of `Isolate`.